build-support/common.sh

```diff
@@ -69,6 +69,11 @@ function determine_python() {
     if [[ "$("${interpreter_path}" --version 2>&1 > /dev/null)" == "pyenv: python${version}"* ]]; then
       continue
     fi
+    # Sometimes the 'python3.Y' binary is a symlink to a suffixed version: 'python3.Ym'.
+    # We use 'realpath' to get the most precise version (the suffixed version or the real file),
+    # or pex can fail to find the interpreter in the venv and then fail to reexecute.
+    # See: https://github.com/pantsbuild/pex/issues/2119
+    interpreter_path="$(realpath "${interpreter_path}")"
```

Comment:
It used to be the case that getting `realpath` on a Mac required some extra action like installing `coreutils` via homebrew. It would be a shame to work around your issue by breaking others. It would be really great to fix the real issue instead. Can you get by with this edit locally if you can't wait for the real fix?

Reply:
Perhaps just use the interpreter path itself to run `-c 'import os.path; ...'`. I think this function is only used in slow paths.

Reply:
I checked for that before I made the PR. I have `/bin/realpath` on my Mac.

If it were installed by macports, it would be in `/opt/local`. I believe homebrew has its own prefix, but I don't use that. Most apps provide CLI utils under the .app directory.

But, googling further, I see that `realpath` was not included in anything older than Mac Ventura: https://apple.stackexchange.com/a/450116

So, you're right, this is not a safe fix.

Reply:
I've got the real fix here: https://github.com/pantsbuild/pex/pull/2122

@cognifloyd if you can wait for the Pex 2.1.133 release later this evening and use it with a custom `pants.toml` `[pex-cli]` setup it seems to me that would be preferable here.

Reply:
Yes, that is preferable. I'll close this. Thank you again for all you did to fix this!